### ISSUE TITLE
deploy kubeflow on juju3

### ISSUE THREAD

**Issue by ycliuhw** [canonical/bundle-kubeflow#464]
Hey guys,
The latest stable kubeflow is broken on Juju3.
Because Juju3 uses jammy (has python 3.10 installed) for the operator image now.
And the envoy charm current is broken on jammy.
```
2022-06-29T17:33:38.314748173Z 2022-06-29 17:33:38 WARNING install Traceback (most recent call last):
2022-06-29T17:33:38.314768603Z 2022-06-29 17:33:38 WARNING install   File "/var/lib/juju/agents/unit-envoy-0/charm/./src/charm.py", line 9, in <module>
2022-06-29T17:33:38.314776734Z 2022-06-29 17:33:38 WARNING install     from envoy_data_plane.envoy.api import v2 as api
2022-06-29T17:33:38.314782854Z 2022-06-29 17:33:38 WARNING install   File "/var/lib/juju/agents/unit-envoy-0/charm/venv/envoy_data_plane/envoy/api/v2/__init__.py", line 10, in <module>
2022-06-29T17:33:38.314795604Z 2022-06-29 17:33:38 WARNING install     from betterproto.grpc.grpclib_server import ServiceBase
2022-06-29T17:33:38.314801425Z 2022-06-29 17:33:38 WARNING install   File "/var/lib/juju/agents/unit-envoy-0/charm/venv/betterproto/grpc/grpclib_server.py", line 2, in <module>
2022-06-29T17:33:38.314844926Z 2022-06-29 17:33:38 WARNING install     from collections import AsyncIterable
2022-06-29T17:33:38.314852796Z 2022-06-29 17:33:38 WARNING install ImportError: cannot import name 'AsyncIterable' from 'collections' (/usr/lib/python3.10/collections/__init__.py)
```
As they suggested here https://github.com/danielgtaylor/python-betterproto/issues/372, we probably should upgrade the `betterproto` to `2.0.0b4`.

Cheers

**Comment by NohaIhab** [comment]
[Jira](https://warthogs.atlassian.net/browse/KF-408?atlOrigin=eyJpIjoiOTZjMmM1ODIxNTQ1NGE0NDg3NjkwNmQxN2NlZmY3YjYiLCJwIjoiaiJ9)

**Comment by DomFleischmann** [comment]
I've been able to deploy the latest bundle of Kubeflow 1.6 on Juju 3 succesfully. Envoy has been removed in this version which seemed to be the only charm with issues. I also was able to [execute](https://charmed-kubeflow.io/docs/ml-workflow-kubeflow-with-katib-and-mlflow) the following workload also. I will close this issue as it seems like Charmed Kubeflow is ready for juju 3. :partying_face:
